testutil/rapidproto/rapidproto.go

```diff
@@ -0,0 +1,228 @@
+package rapidproto
+
+import (
+	"fmt"
+
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+	"gotest.tools/v3/assert"
+	"pgregory.net/rapid"
+)
+
+func MessageGenerator[T proto.Message](x T, options GeneratorOptions) *rapid.Generator[T] {
+	msgType := x.ProtoReflect().Type()
+	return rapid.Custom(func(t *rapid.T) T {
+		msg := msgType.New()
+
+		options.setFields(t, msg, 0)
+
+		return msg.Interface().(T)
+	})
+}
+
+type GeneratorOptions struct {
+	AnyTypeURLs []string
+	Resolver    protoregistry.MessageTypeResolver
+}
+
+const depthLimit = 10
+
+func (opts GeneratorOptions) setFields(t *rapid.T, msg protoreflect.Message, depth int) bool {
+	// to avoid stack overflow we limit the depth of nested messages
+	if depth > depthLimit {
+		return false
+	}
+
+	descriptor := msg.Descriptor()
+	fullName := descriptor.FullName()
+	switch fullName {
+	case timestampFullName:
+		opts.genTimestamp(t, msg)
+		return true
+	case durationFullName:
+		opts.genDuration(t, msg)
+		return true
+	case anyFullName:
+		return opts.genAny(t, msg, depth)
+	case fieldMaskFullName:
+		opts.genFieldMask(t, msg)
+		return true
+	default:
+		fields := descriptor.Fields()
+		n := fields.Len()
+		for i := 0; i < n; i++ {
+			field := fields.Get(i)
+			if !rapid.Bool().Draw(t, fmt.Sprintf("gen-%s", field.Name())) {
+				continue
+			}
+
+			opts.setFieldValue(t, msg, field, depth)
+		}
+		return true
+	}
+}
+
+const (
+	timestampFullName protoreflect.FullName = "google.protobuf.Timestamp"
+	durationFullName                        = "google.protobuf.Duration"
```

Comment:
```suggestion
	durationFullName  protoreflect.FullName = "google.protobuf.Duration"
```
ditto for below lines, add missing type declaration to avoid lint issues